### PR TITLE
xcp: update and simplify interface using fd-send-recv >= 2.0.0

### PR DESCRIPTION
**Pull request by mseri**
Requires a new xs-opam to be released. Do not merge until xs-opam 4.1.0 is released

Signed-off-by: Marcello Seri <marcello.seri@citrix.com>

### CODE REVIEW

File: lib/posix_channel.ml [hunk 1]
```diff
@@ -203,8 +203,9 @@ let receive protocols =
     finally
       (fun () ->
         Unix.connect s (Unix.ADDR_UNIX path);
-        let (_: int) = Unix.send s (Bytes.unsafe_of_string token) 0 (String.length token) [] in
-        let (_, _, fd) = Fd_send_recv.recv_fd s token 0 (String.length token) [] in
+        let token = Bytes.of_string token in
```
[the remark this answers is not in the document]
Comment (mseri) [reply lib/posix_channel.ml]: I don't know. although I believe that it is, but it's definitely unsafe to mutate an immutable string of which we don't know the origin, so we have to copy here

File: lib/posix_channel.ml [hunk 2]
```diff
@@ -203,8 +203,9 @@ let receive protocols =
...
+        let (_: int) = Unix.send s token 0 (Bytes.length token) [] in
```
Comment (mseri) [inline lib/posix_channel.ml]: We could alternatively do `Unix.send_substring` here and `Bytes.create (String.lenght token)` below, as the buffer is kind of throaway as it stands now

File: lib/posix_channel.ml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mseri) [reply lib/posix_channel.ml]: According to line 117 is very short: `let token = "token"`...

File: lib/posix_channel.ml (hunk 2 above)
Comment (gaborigloi) [reply lib/posix_channel.ml]: That sounds even better to me!

**Comment by mseri** [comment]
Ok, channel helper was a thing on its own... and in fact it was never built. It's a sort of proxy, you can run it with `jbuilder exec misc/channel_helper.exe`. This also shows that lwt was completely unnecessary for xcp-idl. I am going to send a PR to xs-opam once we merge this

**Pull request merged**